test/server/config_validation/xds_fuzz.cc

```diff
@@ -0,0 +1,250 @@
+#include "test/server/config_validation/xds_fuzz.h"
+
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/config/endpoint/v3/endpoint.pb.h"
+#include "envoy/config/listener/v3/listener.pb.h"
+#include "envoy/config/route/v3/route.pb.h"
+
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+
+// helper functions to build API responses
+envoy::config::cluster::v3::Cluster XdsFuzzTest::buildCluster(const std::string& name) {
+  return ConfigHelper::buildCluster(name, "ROUND_ROBIN", api_version_);
+}
+
+envoy::config::endpoint::v3::ClusterLoadAssignment
+XdsFuzzTest::buildClusterLoadAssignment(const std::string& name) {
+  return ConfigHelper::buildClusterLoadAssignment(
+      name, Network::Test::getLoopbackAddressString(ip_version_),
+      fake_upstreams_[0]->localAddress()->ip()->port(), api_version_);
+}
+
+envoy::config::listener::v3::Listener XdsFuzzTest::buildListener(uint32_t listener_num,
+                                                                 uint32_t route_num) {
+  std::string name = fmt::format("{}{}", "listener_", listener_num % NUM_LISTENERS);
```

Comment:
very small nit: prefer absl::StrCat over fmt::format? There was an effort to remove the dep, but it's now in C++20. so there seems to be a recommendation for fmt in logs, strcat everywhere else https://github.com/envoyproxy/envoy/issues/9623